hits_ingest.py

```diff
@@ -1,60 +1,105 @@
+or
+$ python hits_ingest.py
+(if you don't want to give it an argument because you're running processCcd)
+
+OUTPUT
+if doIngest: repo populated with *links* to files in datadir, organized by date
+             sqlite3 database registry of ingested images also created in repo
+if doIngestCalibs: sqlite3 database registry of ingested calibration products
+                   created in calibrepo
+if doProcessCcd: outputrepo/visit populated with subdirectories containing the
+                 usual post-ISR data (bkgd, calexp, icExp, icSrc, postISR)
+
+BASH EQUIVALENT
+if doIngest:
+    $ ingestImagesDecam.py repo --filetype raw --mode link datafiles
+if doIngestCalibs:
+    $ ingestCalibs.py repo --calib calibrepo --validity 999 datafiles
+if doProcessCcd:
+    $ cd calibrepo
+    $ processCcd.py ../repo --id visit=visit ccdnum=ccdnum 
+            --output ../outputrepo -C $OBS_DECAM_DIR/config/processCcdCpIsr.py
+            --config calibrate.doAstrometry=False calibrate.doPhotoCal=False
+            --no-versions
 '''
-doIngest = True
-doIngestCalibs = False # doesn't work yet, see below
-
-## edit directory names for ingested data repos here ##
-#repodir = '../ingested/' # on lsst-dev
-#calibrepodir = '../calibingested/' # on lsst-dev
-repodir = 'ingested/' # on laptop
-calibrepodir = 'calibingested/' # on laptop
-
-datadir = sys.argv[1] # or '/lsst7/mrawls/HiTS/Blind15A_38/' on lsst-dev or 'data/' on laptop
-datafiles = glob(datadir + '*.fits.fz')
-
-# first, need to make a text file that handles the mapper, per the obs_decam github README
-f = open(repodir+'_mapper', 'w')
+# edit values below as desired
+doIngest = False
+doIngestCalibs = False
+doProcessCcd = True
+visit = '410877' # used for ProcessCcd only
+ccdnum = '25'    # used for ProcessCcd only
+repo = 'ingested/'
+calibrepo = 'calibingested/'
+outputrepo = 'processed/'
+# edit values above as desired
+
+if ((doIngest and doIngestCalibs) or (doIngest and doProcessCcd) or 
+    (doIngestCalibs and doProcessCcd)):
+    raise RuntimeError('You can only have one task True at a time')
+elif not doIngest and not doIngestCalibs and not doProcessCcd:
+    raise RuntimeError('Nothing to do, every task is set to False')
+
+try:
+    datadir = sys.argv[1]  # '/lsst7/mrawls/HiTS/<dirname>' on lsst-dev
+                       # 'data/' or 'MasterCals/' on laptop
+                       # (if doIngestCalibs, datadir must be the dir of calibs)
+                       # can also use a regex, e.g., 'data/c4d_15*.fz'
+except:
+    datadir = None
+    print('WARNING: no data directory specified.')
+    print('(This is OK if you are just running ProcessCcd.)')
+else:    
+    if datadir[-1] == '/':
+        datafiles = glob(datadir + '*.fits.fz')
+    else:
+        datafiles = glob(datadir)
+
+# make a text file that handles the mapper, per the obs_decam github README
+f = open(repo+'_mapper', 'w')
 print('lsst.obs.decam.DecamMapper', file=f)
 f.close()
 
-if doIngest == True:
+if doIngest:
```

Comment:
Right, I forgot to mention that I'm totally on board with some of these being "out of scope" for this ticket. No worries